handler/src/test/java/io/netty/handler/ssl/SslContextTrustManagerTest.java

```diff
@@ -0,0 +1,148 @@
+        runTests(new String[] { "tm_test_ca_1a.pem", "tm_test_ca_1b.pem", },
+                new String[] { "tm_test_eec_1.pem", "tm_test_eec_2.pem",
+                        "tm_test_eec_3.pem" }, new boolean[] { true, true,
+                        false });
+    }
+
+    @Test
+    public void testUsingCAsOneAandTwo() throws Exception {
+        runTests(new String[] { "tm_test_ca_1a.pem", "tm_test_ca_2.pem" },
+                new String[] { "tm_test_eec_1.pem", "tm_test_eec_2.pem",
+                        "tm_test_eec_3.pem" }, new boolean[] { true, false,
+                        true });
+    }
+
+    /**
+     *
+     * @param caResources
+     *            an array of paths to CA Certificates in PEM format to load
+     *            from the classpath (relative to this class).
+     * @param eecResources
+     *            an array of paths to Server Certificates in PEM format in to
+     *            load from the classpath (relative to this class).
+     * @param expectations
+     *            an array of expecting results for each EEC Server Certificate
+     *            (the array is expected to have the same length the previous
+     *            argument, and be arrange in matching order: true means
+     *            expected to be valid, false otherwise.
+     */
+    private static void runTests(String[] caResources, String[] eecResources,
+            boolean[] expectations) throws Exception {
+        X509TrustManager tm = getTrustManager(caResources);
+
+        X509Certificate[] eecCerts = loadCertCollection(eecResources);
+
+        for (int i = 0; i < eecResources.length; i++) {
+            X509Certificate eecCert = eecCerts[i];
+            assertNotNull("Cannot use cert " + eecResources[i], eecCert);
+            try {
+                tm.checkServerTrusted(new X509Certificate[] { eecCert }, "RSA");
+                if (!expectations[i]) {
+                    fail(String.format(
+                            "Certificate %s was expected not to be valid when using CAs %s, but its "
+                                    + "verification passed.", eecResources[i],
+                            Arrays.asList(caResources)));
+                }
+            } catch (CertificateException e) {
+                if (expectations[i]) {
+                    fail(String.format(
+                            "Certificate %s was expected to be valid when using CAs %s, but its "
+                                    + "verification failed.", eecResources[i],
+                            Arrays.asList(caResources)));
+                }
+            }
+        }
+    }
+
+    private static X509TrustManager getTrustManager(String[] resourceNames)
+            throws Exception {
+        X509Certificate[] certCollection = loadCertCollection(resourceNames);
+        TrustManagerFactory tmf = SslContext.buildTrustManagerFactory(
+                certCollection, null);
+
+        X509TrustManager x509Tm = null;
```

Comment:
This one is not redundant, unless we return the value directly in the loop (instead of breaking) and return null after the loop. Would that be the preferred style?